server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -353,7 +368,10 @@ public void uploadMetadata(Collection<String> segmentFiles, Directory storeDirec
                     throw new NoSuchFileException(file);
                 }
             }
+
+            CodecUtil.writeHeader(indexOutput, UploadedSegmentMetadata.METADATA_CODEC, UploadedSegmentMetadata.CURRENT_VERSION);
             indexOutput.writeMapOfStrings(uploadedSegments);
```

Comment:
can we abstract this method such that it can be linked to the version in which it is being written. That way we can control what was the encoding mechanism for each version?